src/components/graph/util/model-functions.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { UUID } from 'crypto';
+import NetworkModificationTreeModel, {
+    NetworkModificationNode,
+    NodeType,
+    RootNode,
+} from '../network-modification-tree-model';
+import { CurrentTreeNode, TreeNodeData } from 'redux/reducer';
+
+export function convertNodetoReactFlowModelNode(
+    node: NetworkModificationNode | RootNode,
+    parentNodeUuid: UUID | undefined
+): CurrentTreeNode {
+    // Use the type guard to safely access nodeBuildStatus
+    const globalBuildStatus = isNetworkModificationNode(node) ? node.nodeBuildStatus?.globalBuildStatus : undefined;
+    const localBuildStatus = isNetworkModificationNode(node) ? node.nodeBuildStatus?.localBuildStatus : undefined;
+
+    // This is the ReactFlow format (Cf documentation)
+    // {
+    //  id: '1',
+    //  type: 'input',
+    //  data: { label: 'Node 1' }, <- use data for customization
+    //  position: { x: 250, y: 5 }
+    // }
+    const data: TreeNodeData = {
+        parentNodeUuid: parentNodeUuid!,
+        label: node.name,
+        description: node.description ?? null,
+        globalBuildStatus: globalBuildStatus,
+        localBuildStatus: localBuildStatus,
+    };
+
+    return {
+        id: node.id,
+        type: node.type,
+        position: { x: 0, y: 0 },
+        data: data,
+    };
+}
+
+// Return the first node of type nodeType and specific buildStatus
+// in the tree model
+export function getFirstNodeOfType(
+    elements: NetworkModificationNode | RootNode,
+    nodeType: string,
+    buildStatusList: string[] | undefined
+) {
+    return recursiveSearchFirstNodeOfType(
+        elements,
+        undefined, // first is Root node without parent node
+        nodeType,
+        buildStatusList
+    );
+}
+export function isNetworkModificationNode(n: NetworkModificationNode | RootNode): n is NetworkModificationNode {
+    return 'nodeBuildStatus' in n;
+}
+
+// Recursive search of a node of type and buildStatus specified
+export function recursiveSearchFirstNodeOfType(
+    elements: NetworkModificationNode | RootNode,
+    parentNodeUuid: UUID | undefined,
+    nodeType: string,
+    buildStatusList: string[] | undefined
```

Comment:
```suggestion
    parentNodeUuid?: UUID ,
    nodeType: NodeType,
    buildStatusList?: BUILD_STATUS[]
```

Reply:
fixed and i changed the order of the params because of 
TS1016: A required parameter cannot follow an optional parameter.